src/Repeater/Repeater.ts

```diff
@@ -0,0 +1,61 @@
+import { Protocol } from '../RequestExecutor';
+
+export interface RepeaterEvent<I, O = void> {
+  request: I;
+  response: O;
+}
+
+export type RepeaterRequestEvent = RepeaterEvent<
+  {
+    protocol: Protocol;
+    url: string;
+    method?: string;
+    headers?: Record<string, string | string[]>;
+    correlationIdRegex?: string;
+    body?: string;
+  },
+  {
+    protocol: Protocol;
+    statusCode?: number;
+    message?: string;
+    errorCode?: string;
+    headers?: Record<string, string | string[] | undefined>;
+    body?: string;
+  }
+>;
+
+export type RepeaterDeployedEvent = RepeaterEvent<
+  undefined,
+  {
+    repeaterId: string;
+  }
+>;
+
+export type RepeaterReconnectionFailedEvent = RepeaterEvent<{
+  error: Error;
+}>;
+
+export interface RepeaterEvents {
+  deployed: RepeaterDeployedEvent;
+  request: RepeaterRequestEvent;
+  reconnection_failed: RepeaterReconnectionFailedEvent;
+}
+
+export type RepeaterEventHandler<E extends keyof RepeaterEvents> = (
+  payload: RepeaterEvents[E]['request']
+) => RepeaterEvents[E]['response'] | Promise<RepeaterEvents[E]['response']>;
```

Comment:
Avoid using indexed access types as they tend to lack clarity and conciseness, which, in turn, increases cognitive complexity and makes maintenance more challenging. Also, they also may lead to the wrong inheritance violating SOLID.

Reply:
Refactored to function overload approach. But as for me it didn't add much more clarity. Please suggest any better approach if you know one

Reply:
To avoid overloading and provide a clearer and more concise way to handle events, use dedicated methods for event handlers instead of a generic `on` method.

```ts
export interface RepeaterServer {
  // ...
  requestRecevied(handler: RepeaterServerEventHandler<
      RepeaterServerRequestEvent,
      RepeaterServerRequestResponse
    >): void;
  reconnectionFailed(handler: RepeaterServerEventHandler<RepeaterServerReconnectionFailedEvent>): void;
  errorOccurred(handler: RepeaterServerEventHandler<RepeaterServerErrorEvent>): void;
}
```

Also, it is possible to eliminate the complexity, using the `AsyncIterableIterator` as follows:
```ts
export type RepeaterServerRequestAck = (response?: RepeaterServerResponseEvent) => unknown;

export interface RepeaterServer {
  // ...
  requestRecevied(): AsyncIterableIterator<[RepeaterServerRequestEvent, RepeaterServerRequestAck]>;
  reconnectionFailed(): AsyncIterableIterator<RepeaterServerReconnectionFailedEvent>;
  errorOccurred(): AsyncIterableIterator<RepeaterServerErrorEvent>;
}

// ...

for await (const [request, ack] of instance.requestRecevied()) {
   const response = await handleRequest(request);
   ack(response);
}
```

Reply:
Used the first approach, since we have typings for Node.js 10 and `"node": ">=10 <=18"` as engines spec. So node.js 10 lacks some async iteration features like `on` from `events` module.

A also moved `reconnectionAttempted` handler to the launcher and added `reconnectionSucceeded` handler to inform users when they get reconnected

<img width="864" alt="image" src="https://github.com/NeuraLegion/bright-cli/assets/9965996/b2f04aa8-1258-4939-a499-46c22a619c78">